stdlib/public/core/StringBridge.swift

```diff
@@ -427,34 +459,33 @@ private func getConstantTaggedCocoaContents(_ cocoaString: _CocoaString) ->
 
   let taggedValue = unsafeBitCast(cocoaString, to: UInt.self)
   
-  //11000000..payload..111
-  let tagMask:UInt =
-    0b1111_1111_1000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0111
-  let expectedValue:UInt =
-    0b1100_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0111
 
-  guard taggedValue & tagMask == expectedValue else {
-    return nil
-  }
 
-  guard _swift_stdlib_dyld_is_objc_constant_string(
-    cocoaString as! UnsafeRawPointer
-  ) == 1 else {
+  guard taggedValue & constantTagMask == expectedConstantTagValue else {
     return nil
   }
 
-  let payloadMask = ~tagMask
+  let payloadMask = ~constantTagMask
   let payload = taggedValue & payloadMask
   let ivarPointer = UnsafePointer<_swift_shims_builtin_CFString>(
     bitPattern: payload
   )!
+
+  guard _swift_stdlib_dyld_is_objc_constant_string(
+    unsafeBitCast(ivarPointer, to: UnsafeRawPointer.self)
+  ) == 1 else {
+    return nil
+  }
+
   let length = ivarPointer.pointee.length
   let isUTF16Mask:UInt = 0x0000_0000_0000_0004 //CFStringFlags bit 4: isUnicode
   let isASCII = ivarPointer.pointee.flags & isUTF16Mask == 0
+  precondition(isASCII) // we don't currently support non-ASCII here
```

Comment:
What should be the behavior if this check fails? Should we trap in the stdlib?

Reply:
I think so. It's pretty much "assert the world is as we expected"